package/sbp_nmea2000_bridge/src/sbp_nmea2000_bridge.cpp

```diff
@@ -0,0 +1,917 @@
+
+      canfd_frame frame;
+      ssize_t bytes_read = recvfrom(item->fd, &frame,sizeof(frame),
+              /*flags=*/0,
+              /*src_addr=*/NULL,
+              /*addrlen=*/NULL);
+      piksi_check(bytes_read < 0, "Could not read interface %s.",
+                  interface_name);
+
+      std::cout << "Read " << bytes_read << " bytes from " << interface_name
+                <<" ID: " << std::setfill('0') << std::setw(4) << std::hex
+                << frame.can_id
+                << " Data: ";
+      for (u32 i = 0; i < frame.len; ++i) {
+        std::cout << std::setfill('0') << std::setw(2) << std::hex
+                  << static_cast<u16>(frame.data[i]);
+      }
+      std::cout << std::endl;
+
+      return 0;
+    }
+
+    // PGN 126992
+    void callback_sbp_utc_time(u16 sender_id, u8 len, u8 msg[], void *context) {
+      UNUSED(sender_id);
+      UNUSED(len);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n";
+
+      auto sbp_utc_time = reinterpret_cast<msg_utc_time_t*>(msg);
+      u16 days_since_1970;
+      double seconds_since_midnight;
+      calculate_days_and_seconds_since_1970(sbp_utc_time,
+                                            &days_since_1970,
+                                            &seconds_since_midnight);
+      d << "\tTOW: " << sbp_utc_time->tow << "\n"
+        << "\tDays since 1970: " << days_since_1970 << "\n"
+        << "\tSeconds since midnight: " << seconds_since_midnight << "\n";
+
+      bool is_valid = is_flags_valid(sbp_utc_time->flags, /*bits=*/3);
+
+      tN2kMsg N2kMsg;
+      if (is_valid) {
+        SetN2kSystemTime(N2kMsg, tow_sequence_id_cache[sbp_utc_time->tow],
+                         days_since_1970, seconds_since_midnight,
+                         tN2kTimeSource::N2ktimes_GPS);
+      } else {
+        SetN2kSystemTime(N2kMsg, /*sequence ID=*/0xFF,
+                         N2kUInt16NA, N2kDoubleNA,
+                         tN2kTimeSource::N2ktimes_GPS);
+      }
+      NMEA2000.SendMsg(N2kMsg);
+
+      n2k_composite_message_cache.ResetIfOld(sbp_utc_time->tow);
+      n2k_composite_message_cache.SetFields(sbp_utc_time->tow,
+                                            days_since_1970,
+                                            seconds_since_midnight);
+      if (is_valid &&
+          n2k_composite_message_cache.IsReadyToSend(/*is_valid=*/true)) {
+        N2kMsg.Clear();
+        n2k_composite_message_cache.FillN2kMsg(&N2kMsg, /*is_valid=*/true);
+        NMEA2000.SendMsg(N2kMsg);
+      }
+
+      d << "\tDone.\n";
+    }
+
+    // PGN 127250
+    void callback_sbp_baseline_heading(u16 sender_id, u8 len, u8 msg[], void *context){
+      UNUSED(sender_id);
+      UNUSED(len);
+      UNUSED(context);
+      d << __FUNCTION__ << "\n";
+
+      auto sbp_baseline_heading =
+              reinterpret_cast<msg_baseline_heading_t*>(msg);
+
+      d << "\tBaseline heading in deg: "
+        << sbp_baseline_heading->heading / 1000.0 << "\n"
```

Comment:
Please add a constant explaining the 1000.0 value (or store it somewhere with a comment), the computation is repeated three times here.